app/models/goal.py

```diff
@@ -1,5 +1,20 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from typing import Optional
 
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[list["Task"]] = relationship(back_populates="goal")
+
+
+    def to_dict(self):
+        return {
+            "id" : self.id,
+            "title": self.title
+        }
```

Comment:
Nitpick: Add a line between these two methods to visually separate them. 

Per the [PEP8 style guide on blank lines](https://peps.python.org/pep-0008/#blank-lines): "Method definitions inside a class are surrounded by a single blank line."
```suggestion
        }
        
```